modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/coordination/rdbms/RDBMSClusterNotificationListener.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c)2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.andes.server.cluster.coordination.rdbms;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import org.apache.log4j.Logger;
+import org.wso2.andes.configuration.AndesConfigurationManager;
+import org.wso2.andes.configuration.enums.AndesConfiguration;
+import org.wso2.andes.kernel.AndesConstants;
+import org.wso2.andes.kernel.AndesContext;
+import org.wso2.andes.kernel.AndesException;
+import org.wso2.andes.server.ClusterResourceHolder;
+import org.wso2.andes.server.cluster.coordination.ClusterNotificationListener;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * The RDBMSClusterNotificationListener provides means to initialize the node with RDBMS based cluster notification
+ * synchronization..
+ */
+public class RDBMSClusterNotificationListener implements ClusterNotificationListener {
+
+    Logger log = Logger.getLogger(RDBMSClusterNotificationListener.class);
+
+    /**
+     * {@inheritDoc}
+     * <p/>
+     * Schedules the {@link ClusterEventReaderTask} to run with the configured interval in the broker.xml.
+     */
+    @Override
+    public void initializeListener() throws AndesException {
+
+        // Clear cluster notifications stored in the database and schedule a periodic task to read cluster events
+        // from the store if cluster event sync mode is set to RDBMS.
+        if ((AndesContext.getInstance().isClusteringEnabled())) {
+            if (AndesContext.getInstance().getClusterAgent().isCoordinator()) {
+                clearAllClusterNotifications();
```

Comment:
All cluster notification should be cleared only during a cluster restart. Otherwise some messages can be removed before the target node receiving it.